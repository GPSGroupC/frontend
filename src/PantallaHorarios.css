@import url('https://fonts.googleapis.com/css2?family=Roboto:ital,wght@1,300&display=swap');
@import url('https://fonts.googleapis.com/css?family=Raleway');
@import url('https://fonts.googleapis.com/css2?family=Rubik:ital,wght@1,500&display=swap');



#buttongroup {
    padding : 0px;
    float: left;
    border-radius: 0%;
    align-items: center;
    background-color: transparent;
    border: 0px;

}

#buttonselect {
    width: 100%;
    margin: 0px;
    display: block;
    text-align: center;

}

.selected {
    background-color: rgb(126, 124, 124) ;
}

#import-box {
    border: 1px solid black;
    border-radius: 3%;
    float:left;
    margin-left: 10%;
    text-align: center;
    padding: 5%;
}

#importarbutton {
    margin-left: 25px;
}
