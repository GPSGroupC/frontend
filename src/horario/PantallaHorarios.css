/**
 * CSS PARA EL COMPONENTE PANTALLA HORARIOS
 */
.navBar {
    display: flex;
    flex-direction: column;
    width: 40%;
}

.einaImg{
    height: 100px;
    width: 350px;
}


@media only screen and (max-width: 600px) {
    .menuHorarios {
        width: 50%;
        margin:auto;
    }
}
