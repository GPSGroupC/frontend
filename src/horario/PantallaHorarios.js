import React, { useState, Fragment } from "react";
import { nanoid } from "nanoid";
import data from "../mock-data.json";
import FilaReadOnly from "./componentes/FilaReadOnly";
import FilaEditable from "./componentes/FilaEditable";
import { Link } from 'react-router-dom'
import eina from '../images/eina-logo.png'
import '../PantallaHorarios.css';
import Button from 'react-bootstrap/Button';
import ButtonGroup from 'react-bootstrap/Button';

function PantallaHorarios() {

  const [asignaturas, setAsignaturas] = useState(data);
  const [addFormData, setAddFormData] = useState({
    plan: "",
    area: "",
    codigo: "",
    nombre: "",
    curso: "",
    periodo: "",
    horas_estud1: "",
    horas_estud2: "",
    horas_estud3: "",
  });

  const [editFormData, setEditFormData] = useState({
    plan: "",
    area: "",
    codigo: "",
    nombre: "",
    curso: "",
    periodo: "",
    horas_estud1: "",
    horas_estud2: "",
    horas_estud3: "",
  });

  const [editAsignaturaId, setEditAsignaturaId] = useState(null);




  const handleAddFormChange = (event) => {
    event.preventDefault();

    const fieldName = event.target.getAttribute("name");
    const fieldValue = event.target.value;

    const newFormData = { ...addFormData };
    newFormData[fieldName] = fieldValue;

    setAddFormData(newFormData);
  };

  const handleEditFormChange = (event) => {
    event.preventDefault();

    const fieldName = event.target.getAttribute("name");
    const fieldValue = event.target.value;

    const newFormData = { ...editFormData };
    newFormData[fieldName] = fieldValue;

    setEditFormData(newFormData);
  };




  const handleAddFormSubmit = (event) => {
    event.preventDefault();

    const newAsignatura = {
      id: nanoid(),
      plan: addFormData.plan,
      area: addFormData.area,
      codigo: addFormData.codigo,
      nombre: addFormData.nombre,
      curso: addFormData.curso,
      periodo: addFormData.periodo,
      horas_estud1: addFormData.horas_estud1,
      horas_estud2: addFormData.horas_estud2,
      horas_estud3: addFormData.horas_estud3,
    };

    const newAsignaturas = [...asignaturas, newAsignatura];
    setAsignaturas(newAsignaturas);
  };

  const handleEditFormSubmit = (event) => {
    event.preventDefault();

    const editedAsignatura = {
      id: editAsignaturaId,
      plan: editFormData.plan,
      area: editFormData.area,
      codigo: editFormData.codigo,
      nombre: editFormData.nombre,
      curso: editFormData.curso,
      periodo: editFormData.periodo,
      horas_estud1: editFormData.horas_estud1,
      horas_estud2: editFormData.horas_estud2,
      horas_estud3: editFormData.horas_estud3,
     
    };

    const newAsignaturas = [...asignaturas];

    const index = asignaturas.findIndex((asignatura) => asignatura.id === editAsignaturaId);

    newAsignaturas[index] = editedAsignatura;

    setAsignaturas(newAsignaturas);
    setEditAsignaturaId(null);
  };

  const handleEditClick = (event, asignatura) => {
    event.preventDefault();
    setEditAsignaturaId(asignatura.id);

    const formValues = {
      plan: asignatura.plan,
      area: asignatura.area,
      codigo: asignatura.codigo,
      nombre: asignatura.nombre,
      curso: asignatura.curso,
      periodo: asignatura.periodo,
      horas_estud1: asignatura.horas_estud1,
      horas_estud2: asignatura.horas_estud2,
      horas_estud3: asignatura.horas_estud3,
    };

    setEditFormData(formValues);
  };

  const handleCancelClick = () => {
    setEditAsignaturaId(null);
  };

  const handleDeleteClick = (asignaturaId) => {
    const newAsignaturas = [...asignaturas];

    const index = asignaturas.findIndex((asignatura) => asignatura.id === asignaturaId);

    newAsignaturas.splice(index, 1);

    setAsignaturas(newAsignaturas);
  };

  return (
      
    <div>

            <div>
              <Link to="/"><img className="logoCab2" src={eina} /></Link>
              <div className="filtro2" style={{textAlign:'center'}}><br></br><br></br>
              <h4 className="titulo">HORARIOS</h4>
              <Link to="/">
                  <button type="button" className="btn btn-info btn-lg" style={{"margin-left": "750px", "margin-top":"15px"}}>INICIO</button>
              </Link>
              </div>
              <hr size="5px" color="black" />
            </div> <br></br>



            <ButtonGroup vertical id="buttongroup">
                <Link to="/importar-asignaturas"><button id="buttonselect" type="button" >IMPORTAR ASIGNATURAS</button></Link>
                <Link to="/listar-asignaturas"><button id="buttonselect" type="button" >LISTAR ASIGNATURAS</button></Link>
                <Link to="/seleccion-horario-grados"><button id="buttonselect" type="button">EDITAR HORARIOS</button></Link>
            </ButtonGroup>

    </div>
    
  );
};

export default PantallaHorarios;
