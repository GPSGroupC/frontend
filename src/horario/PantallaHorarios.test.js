
import React, { useState, Fragment } from "react";
import ReactDOM from "react-dom";
import PantallaHorarios from "./PantallaHorarios";
import renderer from 'react-test-renderer';
import { BrowserRouter, Route } from 'react-router-dom'



describe("Snapshot testing", () => {

test('Se renderiza correctamente', () => {

    const rendererComponent = renderer.create(<BrowserRouter><PantallaHorarios></PantallaHorarios></BrowserRouter>).toJSON()
    expect(rendererComponent).toMatchSnapshot()

})
})