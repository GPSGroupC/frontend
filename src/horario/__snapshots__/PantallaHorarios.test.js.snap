// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Se renderiza correctamente 1`] = `
<div>
  <div>
    <div
      className="header"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          className="einaImg"
          src="eina-logo.png"
        />
      </a>
      <h4
        className="titulo"
      >
        HORARIOS
      </h4>
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          className="btn btn-info btn-lg"
          type="button"
        >
          INICIO
        </button>
      </a>
    </div>
    <br />
    <hr
      color="black"
      size="5px"
    />
  </div>
  <br />
  <br />
  <div
    style={
      Object {
        "marginLeft": "10%",
      }
    }
  >
                        
    <div
      className="btn-group-toggle navBar"
      data-toggle="buttons"
    >
      <button
        className="btn btn-secondary"
        onClick={[Function]}
        style={
          Object {
            "borderColor": "#e3f2fd",
          }
        }
        type="button"
        value={1}
      >
        IMPORTAR ASIGNATURAS
      </button>
      <button
        className="btn btn-secondary"
        onClick={[Function]}
        style={
          Object {
            "borderColor": "#e3f2fd",
          }
        }
        type="button"
        value={2}
      >
        LISTAR ASIGNATURAS
      </button>
      <button
        className="btn btn-secondary"
        onClick={[Function]}
        style={
          Object {
            "borderColor": "#e3f2fd",
          }
        }
        type="button"
        value={3}
      >
        IMPORTAR AULAS
      </button>
      <button
        className="btn btn-secondary"
        onClick={[Function]}
        style={
          Object {
            "borderColor": "#e3f2fd",
          }
        }
        type="button"
        value={4}
      >
        LISTAR AULAS
      </button>
      <button
        className="btn btn-secondary"
        onClick={[Function]}
        style={
          Object {
            "borderColor": "#e3f2fd",
          }
        }
        type="button"
        value={5}
      >
        LISTAR HORARIOS
      </button>
    </div>
    <br />
    <br />
    <div>
      <div
        id="import-box"
      >
        <h3>
          Importar Asignaturas
        </h3>
        <br />
        <br />
        <p>
          Selecciona el archivo excel para importar las asignaturas
        </p>
        <br />
        <br />
        <div
          id="fileform"
        >
          <form
            onSubmit={[Function]}
          >
            <label>
              <input
                id="file"
                name="file"
                type="file"
              />
            </label>
            <input
              id="importarbutton"
              type="submit"
              value="Importar"
            />
          </form>
          <br />
          <br />
          <label>
            <input
              id="conservarAsignaturas"
              type="checkbox"
            />
            Conservar asignaturas que se habían añadido manualmente
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
